common/sgx/iohandler/file_io.cpp

```diff
@@ -0,0 +1,152 @@
+/* Copyright 2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <string>
+#include <stdlib.h>
+#include <stdio.h>
+#include <iostream>
+#include <stdint.h>
+
+#include "file_io.h"
+#include "iohandler_enclave.h"
+#include "enclave_utils.h"
+
+// Returns iohandler id corresponding to iohandler name
+uint32_t FileIoExecutor::GetIoHandlerId(const char* handler_name) {
+    return TcfGetIoHandlerId(handler_name);
+}
+
+// Maximum size of buffer that is used for file io
+size_t FileIoExecutor::GetMaxFileSize() {
+    return 1024;
+}
+
+// Maximum size of result buffer that is used to store status of io
+size_t FileIoExecutor::GetMaxIoResultSize() {
+    return 128;
+}
+
+/*
+  Opens given file and updates status in result buffer
+  file_name - name of the file
+  result - status of file open operation
+  result_size - Maximum size of the result buffer
+
+*/
+uint32_t FileIoExecutor::FileOpen(uint32_t handler_id, std::string file_name,
+    uint8_t *result, size_t result_size) {
+    uint32_t status;
+    std::string file_operation = "open";
+    std::string command = file_operation + " " + file_name;
+    status = TcfExecuteIoCommand(handler_id, (const uint8_t *)command.c_str(),
```

Comment:
workload or enclave can't do file check because system calls are not supported in SGX. But file existence check is added in untrusted section of file i/o api implementations in file_io_processor.cpp